Gems/Multiplayer/Code/CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ ly_add_target(
             AZ::AzCore
             AZ::AzFramework
             AZ::AzNetworking
+            Legacy::CryCommon
```

Comment:
we really do not want to add dependencies to CryCommon, maybe use a more recent system?

Reply:
Multiplayer gem was already relying on CryCommon, I just moved in the cmake so I could use it for this purpose :c

Reply:
Fixed! Updated to use the new ComponentApplicationLifecycle